packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -299,13 +299,16 @@ export class KeybindingWidget extends ReactWidget {
         }
     }
 
-    protected validateKeybinding(command: string, keybinding: string): string {
+    protected validateKeybinding(command: string, oldKeybinding: string, keybinding: string): string {
         if (!keybinding) {
             return 'keybinding value is required';
         }
         try {
             const binding = { 'command': command, 'keybinding': keybinding };
             KeySequence.parse(keybinding);
+            if (oldKeybinding === keybinding) {
+                return ' '; // if old and new keybindings match, quietly reject update
+            }
```

Comment:
Validation:
1- If I enter the same value, the "OK" button is not available, but If I enter another key definition, then the "OK is available --> confusion to end user . In this case, may be we should let the user knows he modified the keybinding with the original value if the OK button is not available.
2- If I enter a valid key which is defined for another command, we should not allow it under the same context
3- If I only put "alt" for a command, it is not valid, otherwise it would invalidate all commands with "alt", so the "ok" button should not be available in this case

Reply:
@lmcbout 
1. The idea is that the validation will only be successful if an update to the `keybinding`. If the current value and new value match, I don't want to perform an unnecessary update.
2. `keybindings` are only unique for  a given `context` and `scope`. If a keybinding is created for the same context, the one that is present with a `user` scope, will shadow that of the `default` one. easier to view with #3222 
3. I agree that certain keys may not be valid, but this will require a change to theia's `core` api and should be fixed in a seperate pull-request.